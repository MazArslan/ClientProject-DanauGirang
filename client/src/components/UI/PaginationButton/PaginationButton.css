.arrow-button {
	margin-left: 20px;
	margin-right: 20px;
	margin-top: 6px;
	margin-bottom: 6px;
}
