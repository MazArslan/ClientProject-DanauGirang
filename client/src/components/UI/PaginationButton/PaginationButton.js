import React from "react";
import propTypes from "prop-types";
import { Button } from "reactstrap";
import "./PaginationButton.css";

/**
 * Specialised button component for allowing users to navigate between image card components
 * @param { isDisabled, clicked, text } props
 */
const paginationButton = props => {
  const { isDisabled, clicked, text } = props;
  return (
    <Button
      className="arrow-button"
      disabled={isDisabled}
      outline
      color="info"
      onClick={clicked}
    >
      {text}
    </Button>
  );
};

paginationButton.propTypes = {
  isDisabled: propTypes.bool.isRequired,
  clicked: propTypes.func.isRequired,
  text: propTypes.string.isRequired
};

export default paginationButton;
