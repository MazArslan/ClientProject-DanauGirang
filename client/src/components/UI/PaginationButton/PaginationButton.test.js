import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import PaginationButton from "./PaginationButton";

Enzyme.configure({ adapter: new Adapter() });

describe("<PaginationButton/>", () => {
  const onActionMock = jest.fn();
  const props = {
    isDisabled: false,
    clicked: onActionMock,
    text: ">"
  };
  const container = shallow(<PaginationButton {...props} />);

  it("should render correctly", () => {
    expect(container.exists()).toBe(true);
    expect(container.html()).toMatchSnapshot();
  });

  it("should render text", () => {
    const text = ">";
    expect(container.props("text").children).toEqual(text);
  });

  it("should render button", () => {
    expect(container.find("Button").props()).toEqual({
      children: ">",
      className: "arrow-button",
      color: "info",
      disabled: false,
      onClick: onActionMock,
      outline: true,
      tag: "button"
    });
  });
});
