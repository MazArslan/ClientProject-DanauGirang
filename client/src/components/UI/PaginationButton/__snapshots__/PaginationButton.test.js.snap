// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaginationButton/> should render correctly 1`] = `"<button type=\\"button\\" class=\\"arrow-button btn btn-outline-info\\">&gt;</button>"`;
